util/annotations/helpers.go

```diff
@@ -58,6 +58,11 @@ func HasWithPrefix(prefix string, annotations map[string]string) bool {
 	return false
 }
 
+// ReplicasManagedByExternalAutoscaler returns true if the standard annotation for external autoscaler is present.
+func ReplicasManagedByExternalAutoscaler(o metav1.Object) bool {
+	return hasAnnotationValue(o, clusterv1.ReplicasManagedByAnnotation, clusterv1.ExternalAutoscalerReplicasManagedByAnnotationValue)
```

Comment:
what's the benefit of enforcing a specific annotation value for this as opposed to just looking for the annotation to be present? One advantage I can think of for not doing it is that it would allow providers to set more specific values to inform the user about which autoscaler is used.

Reply:
This is a result of this conversation:

https://github.com/kubernetes-sigs/cluster-api/pull/7107#pullrequestreview-1167312243

Reply:
@vincepri's suggestion was 

> And external-autoscaler could be one of the values of the map, the code would only check for the presence of the annotation though

Reply:
I think I'm -1 on that idea. If we are going to put all of the semantic significance into the key name, then boolean is (IMO) appropriate. If the semantic significance is shared between key (replicas managed by...) and value (...some thing that manages replicas) then we should assume the semantic significance in the value string is important.

Reply:
After discussion in today's capi office hours, @vincepri clarified why a "truthy" boolean annotation is the most idiomatic solution. I've made that change, PTAL!

Reply:
> After discussion in today's capi office hours, @vincepri clarified why a "truthy" boolean annotation is the most idiomatic solution. I've made that change, PTAL!

+1, i think Vince's argument was convincing as well.

Reply:
+1